File: user/mods/Gekos_BetterProgression/src/miscChanges/firChanges.ts
import { IStageRequirement } from "@spt/models/eft/hideout/IHideoutArea";
import { Context } from "../contex";
import { DependencyContainer } from "tsyringe";
import { RepeatableQuestGenerator } from "@spt/generators/RepeatableQuestGenerator";
import { IQuestCondition } from "@spt/models/eft/common/tables/IQuest";

export function removeFirFromQuests(context: Context): void
{
    const quests = context.tables.templates.quests;
    const locales = context.tables.locales.global;
    const foundInRaidRegex = new RegExp("Find.*in raid", "i");
    const inRaidRegex = new RegExp("in raid", "i");

    for (const quest of Object.values(quests))
    {
        const sets = [quest.conditions.AvailableForFinish, quest.conditions.AvailableForStart, quest.conditions.Fail, quest.conditions.Started, quest.conditions.Success]

        for (const set of sets)
        {
            if (set == null) continue;

            //Allow handing over of non-FiR items
            for (const cond of set)
            {
                if (cond.conditionType == "HandoverItem" || cond.conditionType === "FindItem")
                {
                    cond.onlyFoundInRaid = false;
                }
            }
        }
    }

    //Change the locales to remove "in raid"
    for (const [lang, locale] of Object.entries(locales)) for (const [key, text] of Object.entries(locale))
    {
        if (foundInRaidRegex.test(text))
        {
            locales[lang][key] = text.replace(inRaidRegex, "");
        }
    }
}

export function removeFirFromFlea(context: Context): void
{
    context.tables.globals.config.RagFair.isOnlyFoundInRaidAllowed = false;
}

export function removeFirFromHideout(context: Context): void
{

    const hideoutAreas = context.tables.hideout.areas;

    for (const area of hideoutAreas) for (const stage of Object.values(area.stages))
    {
        const itemReq = stage.requirements.filter((req) => req.type == "Item");

        for (const req of itemReq as IStageRequirement[])
        {
            req.isSpawnedInSession = false;
        }
    }

}

export function removeFirFromRepeatables(context: Context, container: DependencyContainer): void
{
    container.afterResolution("RepeatableQuestGenerator", (_t, result: RepeatableQuestGenerator) =>
    {
        // We want to replace the original method logic with something different
        const old = (result as any).generateCompletionAvailableForFinish.bind(result);
        (result as any).generateCompletionAvailableForFinish = (itemTpl: string, value: number) =>
        {
            const condition: IQuestCondition = old(itemTpl, value); //Add old logic back in

            try
            {
                condition.onlyFoundInRaid = false;
            
                return condition;
            }
            catch (error)
            {
                this.context.logger.error("Error Details:" + "Something went wrong when trying set FiR requirement to false for repeatable quests!");
                this.context.logger.error("Stack Trace:\n" + (error instanceof Error ? error.stack : "No stack available"));
                return condition;
            }
        };
        // The modifier Always makes sure this replacement method is ALWAYS replaced
    }, {frequency: "Always"});
}

File: user/mods/Gekos_BetterProgression/src/Changes/FirChanges.cs
using SPTarkov.DI.Annotations;
using SPTarkov.Server.Core.DI;
using SPTarkov.Server.Core.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SPTarkov.Server.Core.Models.Eft.Common.Tables;
using SPTarkov.Server.Core.Models.Eft.Hideout;
using GekosBetterProgression;
using JetBrains.Annotations;
using System.Text.RegularExpressions;
using SPTarkov.Server.Core.Models.Spt.Config;

namespace gekos_server.Changes;

public class FirChanges()
{

    public static void RemoveFirFromQuests(Context context)
    {

        Regex foundInRaidRegex = new Regex("Find.*in raid", RegexOptions.IgnoreCase);
        Regex inRaidRegex = new Regex("in raid", RegexOptions.IgnoreCase);

        foreach (Quest quest in context.tables.Templates.Quests.Values)
        {
            var sets = new List<List<QuestCondition>?> {
                quest.Conditions.AvailableForFinish,
                quest.Conditions.AvailableForStart,
                quest.Conditions.Fail,
                quest.Conditions.Started,
                quest.Conditions.Success
            };

            foreach (var set in sets)
            {
                if (set == null)
                {
                    continue;
                }

                foreach (var condition in set)
                {
                    if (condition.ConditionType == "HandoverItem" || condition.ConditionType == "FindItem")
                    {
                        condition.OnlyFoundInRaid = false;
                    }
                }
            }
        }

        var locales = context.tables.Locales.Global;

        // Remove "in raid" from locale text
        foreach (var lang in locales.Keys)
        {
            var locale = locales[lang].Value;
            if (locale is null) continue;

            foreach (var key in locale.Keys.ToList())
            {
                string text = locale[key];

                if (foundInRaidRegex.IsMatch(text))
                {
                    locale[key] = inRaidRegex.Replace(text, "");
                }
            }
        }
    }

    public static void RemoveFirFromFlea(Context context)
    {
        context.tables.Globals.Configuration.RagFair.IsOnlyFoundInRaidAllowed = false;
    }

    public static void RemoveFirFromHideout(Context context)
    {
        List<HideoutArea> hideoutAreas = context.tables.Hideout.Areas;

        foreach (var area in hideoutAreas)
        {
            foreach (var stage in area.Stages.Values)
            {
                List<StageRequirement>? itemReq = stage.Requirements?.FindAll(item => item.Type == "Item");

                if (itemReq is null)
                {
                    context.logger.Error($"Something went wrong when fetching requirements for {area.Type}");
                    continue;
                }

                foreach (var req in itemReq)
                {
                    req.IsSpawnedInSession = false;
                }
            }
        }
    }

    public static void RemoveFirFromRepeatables(Context context)
    {
        var questConfig = context.sptConfig.GetConfigByString<QuestConfig>("Quest");

        if (questConfig?.RepeatableQuests == null)
        {
            context.logger.Warning("Repeatable quest config not found, skipping FiR removal");
            return;
        }

        foreach (var repeatable in questConfig.RepeatableQuests)
        {
            foreach (var completion in repeatable.QuestConfig.CompletionConfig)
            {
                completion.RequiredItemsAreFiR = false;
            }
        }
    }
}